array-methods.js

```diff
@@ -102,20 +158,80 @@ var stateSums = null;
     round this number to the nearest 10th of a cent before moving on.
   )
  */
-var sumOfHighInterests = null;
+
+var sumOfHighInterests = 0;
+let filteredOtherArray = bankBalances;
+
+
+statesArray.forEach(currentState => {
+  filteredOtherArray = filteredOtherArray.filter(currentAccount => {
+    return (currentAccount.state !== currentState);
+  });
+});
+
+var otherStateSums = {};
+
+filteredOtherArray.forEach(currentAccount => {
+  let balance = parseFloat(currentAccount.amount);
+  let state = currentAccount.state;
+
+  if (!otherStateSums[state]) {
+    otherStateSums[state] = 0.0;
+  }
+
+  otherStateSums[state] = roundAndAdd(otherStateSums[state], balance);
+
+});
+
+sumOfHighInterests = Object.values(otherStateSums)
+  .reduce((sum, currentStateTotal) => {
+    let interest = parseFloat(currentStateTotal) * 0.189;
+    if (interest > 50000) {
+      sum = roundAndAdd(sum + interest);
+    }
+
+    return sum;
+
+  }, 0);
```

Comment:
It can help with clarity to give each iterator method its own responsibility.

for sumOfHighInterests you can break it down to the following steps.
```js
Object.values(otherStateSums)
  .map() //calculate the interest here
  .filter() //filter out all values less than 50k
  .reduce() //aggregate the remaining values to a single sum.
```